### PR TITLE
refactor genesis block, add logic serialize to socket, refactor seria…

### PR DESCRIPTION
**Pull request by Verbalist**
…lizer, fix genesis generator

https://trello.com/c/p9EqFoke/141-add-serialize-logic-to-socket

### CODE REVIEW

File: core/controller/transaction.ts [hunk 1]
```diff
@@ -8,22 +8,24 @@ import TransactionPool from 'core/service/transactionPool';
...
             return;
         }
 
-        if (TransactionPool.has(data.trs)) {
+        if (TransactionPool.has(trs)) {
             return;
         }
 
-        const trs = new Transaction(data.trs);
+        // const trs = new Transaction(data.trs);
```
Comment (axelsheva) [inline core/controller/transaction.ts]: Does need remove?

File: core/repository/block/pg.ts [hunk 2]
```diff
@@ -59,7 +59,8 @@ class BlockPGRepo implements IBlockPGRepository {
             fee: rawBlock.fee,
             payloadHash: rawBlock.payload_hash,
             generatorPublicKey: rawBlock.generator_public_key,
-            signature: rawBlock.signature
+            signature: rawBlock.signature,
+            transactions: []
```
Comment (axelsheva) [inline core/repository/block/pg.ts]: when we will push transactions to block?

File: core/controller/transaction.ts [hunk 3]
```diff
@@ -8,22 +8,24 @@ import TransactionPool from 'core/service/transactionPool';
 import { Account } from 'shared/model/account';
 import AccountRepo from 'core/repository/account';
 import { ed } from 'shared/util/ed';
+import TransactionPgRepo from 'core/repository/transaction/pg';
 
 class TransactionController extends BaseController {
     @ON('TRANSACTION_RECEIVE')
     public async onReceiveTransaction(action: { data: { trs: TransactionModel<IAsset> } }): Promise<void> {
         const { data } = action;
+        const trs = TransactionPgRepo.deserialize(data.trs);
```
Comment (axelsheva) [inline core/controller/transaction.ts]: I think here should be TransactionRepo

File: core/genesisGenerator.ts [hunk 4]
```diff
@@ -57,7 +58,7 @@ const rawTransactions: Array<{trs: TransactionModel<IAsset>, privateKey: string}
                 recipientAddress: BigInt('933553974927686133')
             }
         },
-        privateKey: 'hen worry two thank unfair salmon smile oven gospel grab latin reason'
```
Comment (axelsheva) [inline core/genesisGenerator.ts]: It is not a private key, it is secret

File: core/repository/block/pg.ts (hunk 2 above)
Comment (Verbalist) [reply core/repository/block/pg.ts]: after deserialize

File: core/genesisGenerator.ts (hunk 4 above)
Comment (Verbalist) [reply core/genesisGenerator.ts]: +

Comment (Verbalist) [reply core/genesisGenerator.ts]: I not fully rewrite it

**Pull request merged**